hack/camel-crds/main.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+// Script to copy generated CRD yaml into chart directory and modify it to match
+// the expected chart format (e.g. formatted YAML).
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"sigs.k8s.io/yaml"
+)
+
+func main() {
+	if len(os.Args) != 1 {
+		fmt.Println("Usage: go run ./...")
+		os.Exit(1)
+	}
+
+	if err := realMain(); err != nil {
+		fmt.Printf("Error: %s\n", err)
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func realMain() error {
+	root := "../../control-plane/config/crd/"
+	dirs := []string{"bases"}
+
+	for _, dir := range dirs {
+		err := filepath.Walk(root+dir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() || filepath.Ext(path) != ".yaml" || filepath.Base(path) == "kustomization.yaml" {
+				return nil
+			}
+			printf("processing %s", filepath.Base(path))
+
+			contentBytes, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			jsonBytes, err := yaml.YAMLToJSON(contentBytes)
+			if err != nil {
+				return err
+			}
+			fixedJsonBytes := convertKeys(jsonBytes)
+			contentsCamel, err := yaml.JSONToYAML(fixedJsonBytes)
```

Comment:
~ Not a blocker, but wondering whether direct YAML manipulation via a similar library like `go-yaml` would let us do the same thing w/o an intermediate JSON conversion

Reply:
i remember attempting that. but there is no equivalent of json.RawMessage which makes it difficult (i spent a lot of time and couldnt find an alternative) to perform this without the intermediate conversion. IMO since we only have to do this once per file, and it's a make step, it doesnt matter but im happy to improve this solution and make it cleaner.